### PR TITLE
code updated and view done

### PR DESCRIPTION
**Pull request by sumedh-mohod**
data inserted and updated and view done and labales for details and select for university id remain.

### CODE REVIEW

**Comment by sumedh-mohod** [comment]
select option for university id and course major added and deleted funtion added

File: app/Http/Controllers/CoursesController.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        return view('courseShow', compact('show'));
+    }
+
+    public function show(Request $request)
+    {
+        $courseInformation = Course::where('id', $request->id)->get()->first();
+        $universities = University::get();
+        $course_major = CourseMajor::get();
+        $user = [];
+        // dd($courseInformation);
```
Comment (ithqindia) [inline app/Http/Controllers/CoursesController.php:50]: Not required. Remove

File: app/Models/Country.php [hunk 2]
```diff
@@ -11,7 +11,7 @@ class Country extends Model
     use HasFactory;
 
     protected $guarded = [];
-
+    protected $table = "courses";
```
Comment (ithqindia) [inline app/Models/Country.php:14]: Remove this line. Not required as model Country plural is countries hence Laravel automatically gets the table name as countries

File: app/Models/Course.php [hunk 3]
```diff
@@ -11,7 +11,7 @@
 class Course extends Model
 {
     use HasFactory;
-
+    protected $table="courses";
```
Comment (ithqindia) [inline app/Models/Course.php:14]: Remove

File: routes/web.php [hunk 4]
```diff
@@ -153,6 +166,19 @@
...
+Route::post('/courses', [CoursesController::class, 'store']);
+Route::get('/courses/{id}', [CoursesController::class, 'show']);
+Route::get('/courses', [CoursesController::class, 'index']);
+Route::put('/courses', [CoursesController::class, 'update']);
+Route::get('/courses/{id}/view', [CoursesController::class, 'view']);
+Route::get('/courses', function () {
+    $universities = University::get()->sortBy('name');
+    $course_major = CourseMajor::get()->sortBy('name');
+    return view('courses', compact('universities','course_major'));
+});
```
Comment (ithqindia) [inline routes/web.php:179]: remove this. It should be handled in Route::get('/courses', [CoursesController::class, 'index']);

**Review by ithqindia — CHANGES_REQUESTED** [review]
Format all files

File: app/Http/Controllers/CoursesController.php [hunk 5]
```diff
@@ -13,21 +13,16 @@ class CoursesController extends Controller
 {
     public function index()
     {
-        $Course = Course::get();
-        return view('courses', compact('Course'));
+        $courses = Course::get();
+        $universities = University::get()->sortBy('name');
+        $course_major = CourseMajor::get()->sortBy('name');
```
Comment (ithqindia) [inline app/Http/Controllers/CoursesController.php:18]: Variable name $courseMajors

File: app/Http/Controllers/CoursesController.php [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        Course::create([
+            'name' => $request->get('name'),
+            'details' => $request->get('details'),
+            'university_id' => $request->get('university_id'),
+            'course_major_id' => $request->get('course_major_id'),
+            'is_enabled' => (bool)$request->get('is_enabled'),
+        ]);
+        $courseShow = Course::get();
+        Session::flash('message', 'Data added successfully !');
+        return view('course-show', compact('courseShow'));
```
Comment (ithqindia) [inline app/Http/Controllers/CoursesController.php:37]: return redirect....

File: app/Http/Controllers/CoursesController.php [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    public function show(Request $request)
+    {
+        $courseInformation = Course::where('id', $request->id)->get()->first();
+        $universities = University::get();
+        $course_major = CourseMajor::get();
```
Comment (ithqindia) [inline app/Http/Controllers/CoursesController.php:44]: $courseMajors

File: resources/views/course-show.blade.php [hunk 8]
```diff
@@ -23,15 +23,17 @@
             <th> Enabled</th>
             <th> Action</th>
         </tr>
-        @foreach ($show as $course)
+        @php $counter = 1; @endphp
```
Comment (ithqindia) [inline resources/views/course-show.blade.php:26]: Wrong

File: resources/views/course-show.blade.php [hunk 9]
```diff
@@ -23,15 +23,17 @@
...
+        @foreach ($courseShow as $course)
             <tr class="text-center">
-                <td> {{ $course->id }}</td>
+                <td> {{ $counter }}</td>
```
Comment (ithqindia) [inline resources/views/course-show.blade.php:29]: https://laravel.com/docs/8.x/blade#the-loop-variable

**Review by ithqindia — CHANGES_REQUESTED** [review]

**Comment by sumedh-mohod** [comment]
code updated but showing some error in form

File: app/Http/Controllers/CoursesController.php [hunk 10]
```diff
@@ -34,43 +40,51 @@ public function store(Request $request)
...
+{
+    $courseEdit = Course::find($id);
+    $actionUrl = '/courses/ . $id';
+    return view('courses-form', compact('courseEdit','actionUrl'));
+}
+
+    public function update(Request $request, $id)
     {
-        $courseInformation = Course::find($request->id);
+        $courseInformation = Course::find($id);
```
Comment (ithqindia) [inline app/Http/Controllers/CoursesController.php:64]: name it as simple $course

File: routes/web.php [hunk 11]
```diff
@@ -96,9 +96,9 @@
     return view('post-graduation-details', compact('users'));
 });
 
-Route::get('/course-show', function () {
+Route::get('/courses', function () {
```
Comment (ithqindia) [inline routes/web.php:99]: This route is no longer required

**Review by ithqindia — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/CoursesController.php [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+    public function index()
+    {
+        $courses = Course::get();
+        $universities = University::get()->sortBy('name');
+        $courseMajors = CourseMajor::get()->sortBy('name');
+        return view('courses', compact('courses', 'universities', 'courseMajors'));
+    }
+
+    public function create()
+{
```
Comment (ithqindia) [inline app/Http/Controllers/CoursesController.php:23]: Format this file

File: app/Http/Controllers/CoursesController.php [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+        Course::create([
+            'name' => $request->get('name'),
+            'details' => $request->get('details'),
+            'university_id' => $request->get('university_id'),
+            'course_major_id' => $request->get('course_major_id'),
+            'is_enabled' => (bool)$request->get('is_enabled'),
+        ]);
+        $courses = Course::get();
+        Session::flash('message', 'Data added successfully !');
+        return redirect('courses', compact('courseShow'));
```
Comment (ithqindia) [inline app/Http/Controllers/CoursesController.php:43]: return redirect('courses');

File: app/Http/Controllers/CoursesController.php [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+    {
+        Course::find($id)->delete();
+        Session::flash('message', 'Data deleted successfully !');
+        return redirect('/courses');
+    }
+    // public function view(Request $request)
+    // {
+    //     $course = Course::find($request->id);
+    //     return view('course-info', compact('course'));
+    // }
```
Comment (ithqindia) [inline app/Http/Controllers/CoursesController.php:89]: delete these lines

File: resources/views/courses-form.blade.php [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+            </div> --}}
+            {{-- <div class="col-md-12 my-5">
+                <label for="inputcourses_name" class="form-label">Course major ID</label>
+                <select name="course_major_id" class="form-control">
+                    @foreach ($courseMajors as $major)
+                        <option value="{{ $major->id }}" @if (isset($course) && $course->course_major_id == $major->id) selected @endif> {{ $major->id }}-
+                            {{ $major->name }}</option>
+                    @endforeach
+                </select>
+            </div> --}}
```
Comment (ithqindia) [inline resources/views/courses-form.blade.php:49]: Remove commented code

File: resources/views/includes/sidebar/menu.blade.php [hunk 16]
```diff
@@ -86,6 +86,16 @@
...
+            <div class="menu-item">
+                <a class="menu-link" href="/courses">
+                    <span class="menu-title">Courses form</span>
+                </a>
+            </div>
+            <div class="menu-item">
+                <a class="menu-link" href="/course-show">
+                    <span class="menu-title">Course show</span>
+                </a>
+            </div>
```
Comment (ithqindia) [inline resources/views/includes/sidebar/menu.blade.php:98]: This is not required

File: routes/web.php [hunk 17]
```diff
@@ -92,6 +96,9 @@
     return view('post-graduation-details', compact('users'));
 });
 
+Route::get('/delete', function () {
+    return view('delete');
+});
```
Comment (ithqindia) [inline routes/web.php:101]: Delete this

File: routes/web.php [hunk 18]
```diff
@@ -153,6 +160,16 @@
...
 Route::get('/post-graduation-details/{id}/view', [EducationalInformationPostgraduationController::class, 'view']);
 
+// courses
+Route::get('/courses', [CoursesController::class, 'index']);
+Route::get('/courses/add', [CoursesController::class, 'create']);
+Route::post('/courses', [CoursesController::class, 'store']);
+Route::get('/courses/{id}', [CoursesController::class, 'show']);
+Route::get('/courses/{id}/edit', [CoursesController::class, 'edit']);
+Route::put('/courses/{id}', [CoursesController::class, 'update']);
+// Route::get('/courses/{id}/view', [CoursesController::class, 'view']);
```
Comment (ithqindia) [inline routes/web.php:170]: Delete commented line

**Review by ithqindia — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/CoursesController.php (hunk 12 above)
Comment (ithqindia) [reply app/Http/Controllers/CoursesController.php:23]: This is not done

**Review by ithqindia — CHANGES_REQUESTED** [review]

**Review by ithqindia — APPROVED** [review]

**Comment by ithqindia** [comment]
Merged locally

**Pull request closed without merging**